runtime/gc_glue_java/EnvironmentDelegate.hpp

```diff
@@ -159,6 +169,39 @@ class MM_EnvironmentDelegate
 
 	void forceOutOfLineVMAccess();
 
+	MMINLINE void
+	preMove(MM_GCExtensionsBase *extensions, omrobjectptr_t objectPtr)
+	{
+
+		bool hashed = extensions->objectModel.hasBeenHashed(objectPtr);
```

Comment:
perhaps you can even cache ObjectModel (and not Ext as previously suggested)?

_language_vm that you need later you can get from already cached Env.